### ISSUE TITLE
Doxygen deprecated PERL_PATH and MSCGEN_PATH

### ISSUE THREAD

**Issue by jennuine** [ignitionrobotics/ign-cmake#122]
Was compiling a Dome workspace on Focal with Doxygen version 1.8.17 and almost all packages generated a warning similar to this: 

```bash
warning: Tag 'PERL_PATH' at line 2125 of file '/home/osrf/Workspaces/ignDome_ws/build/ignition-launch3/api_tagfile.dox' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
warning: Tag 'MSCGEN_PATH' at line 2147 of file '/home/osrf/Workspaces/ignDome_ws/build/ignition-launch3/api_tagfile.dox' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
```

the packages that this happens for are:

```
12 packages had stderr output: ignition-common3 ignition-fuel_tools5 ignition-gazebo4 ignition-gui4 ignition-launch3 ignition-math6 ignition-msgs6 ignition-physics3 ignition-plugin1 ignition-rendering4 ignition-sensors4 ignition-transport9
```

Based on [Doxygen's changelog](https://www.doxygen.nl/manual/changelog.html), `PERL_PATH` and `MSCGEN_PATH` have been removed in 1.8.16

**Comment by chapulina** [comment]
It should probably be safe to remove them, I don't think we're using Perl, and mcsgen isn't even set:

https://github.com/ignitionrobotics/ign-cmake/blob/a4c8568a4ef91b0abc6c64a50185721f2fd1a33d/doc/doxygen/api.in#L2112-L2116

https://github.com/ignitionrobotics/ign-cmake/blob/a4c8568a4ef91b0abc6c64a50185721f2fd1a33d/doc/doxygen/api.in#L2131-L2138

**Comment by jennuine** [comment]
This would have to be removed for all 12 packages as well (e.g., ign-common, ign-gazebo, etc)

**Comment by chapulina** [comment]
Luckily most packages are just using the doxygen config from `ign-cmake`, so I think that removing those here would remove most of the warnings.

I only see those variables in a few other packages:

https://github.com/search?q=org%3Aignitionrobotics+PERL_PATH&type=code

* `ign-rndf` isn't actively maintained, so we can ignore it
* `ign-tools` doesn't use `ign-cmake`, so we need to update it
* `ign-fuel-tools` uses `ign-cmake`, so I think it shouldn't have that file, we should look into removing that

**Comment by jennuine** [comment]
I tested removing `PERL_PATH` and `MSCGEN_PATH` from [ign-cmake/doc/doxygen/api.in](https://github.com/ignitionrobotics/ign-cmake/blob/ign-cmake2/doc/doxygen/api.in) on a clean build and do not receive any errors from `ign-tools`, `ign-fuel-tools`, or any other package except for `ign-gui`:
```
--- stderr: ignition-gui4                                 
error: Illegal format for option TAGFILES, no name specified for item '='
error: Illegal format for option TAGFILES, no name specified for item '='
error: Illegal format for option TAGFILES, no name specified for item '='
error: Illegal format for option TAGFILES, no name specified for item '='
error: Illegal format for option TAGFILES, no name specified for item '='
error: Illegal format for option TAGFILES, no name specified for item '='
error: Illegal format for option TAGFILES, no name specified for item '='
error: Illegal format for option TAGFILES, no name specified for item '='
---

```

**Comment by chapulina** [comment]
Interesting, here's where `TAGFILES` is defined for `ign-gui`:

https://github.com/ignitionrobotics/ign-gui/blob/12198ffae7fffd96d155f71c8f86fb96063bf0ab/CMakeLists.txt#L114-L120

**Comment by chapulina** [comment]
Fixed in #136.

I'm closing this, but feel free to open a new one for the `TAGFILES`, @jennuine .
